modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3RetryingInputStreamTests.java

```diff
@@ -81,24 +86,65 @@ public void testRangeInputStreamIsAborted() throws IOException {
         assertThat(stream.isAborted(), is(true));
     }
 
+    public void testReadAfterBlobLengthThrowsRequestedRangeNotSatisfiedException() throws IOException {
+        final byte[] bytes = randomByteArrayOfLength(randomIntBetween(1, 512));
+        {
+            int position = bytes.length;
+            int length = randomIntBetween(1, 100);
+            try (var stream = createInputStream(bytes, position, length)) {
+                assertThat(Streams.consumeFully(stream), equalTo(0L));
+            }
+        }
+        {
+            final int position = bytes.length + randomIntBetween(1, 100);
+            final int length = randomIntBetween(1, 100);
+            var exception = expectThrows(RequestedRangeNotSatisfiedException.class, () -> {
+                try (var ignored = createInputStream(bytes, position, length)) {
+                    throw new AssertionError("Should have failed");
```

Comment:
Can we use `fail()` instead?

Reply:
Fixed in ada72f1d01b165090d4bec76ee7d6fc8d35f1628